components/MultiPanel/index.ts

```diff
@@ -1,12 +1,197 @@
+
   constructor () {
     super();
+    
+    // add EventListners
+    this.addEventListener('click', this._onClick);
+    this.addEventListener('keydown', this._onKeyDown);
+
+    // Watch for children changes.
+    new MutationObserver(() => this._childrenChange())
+      .observe(this, { childList: true });
   }
 
   connectedCallback () {
+    this._childrenChange();
+  }
+  
+  // Click event handler
+  _onClick(event:Event) {
+    const el: Element = event.target;
+    this._expand(this._getClosestHeading(el));
+  }
+
+  // KeyDown event handler
+  _onKeyDown(event) {
+    const selectedEl : Element = event.target;
+    
+    // if keydown event is not on heading element, ignore
+    if (!this._getClosestHeading(selectedEl)) {
+      return;
+    }
+
+    // don’t handle modifier shortcuts used by assistive technology.    
+    if (event.altKey) {
+      return;
+    }
+    let newHeading;
+    switch (event.key) {
+      case 'ArrowLeft':
+      case 'ArrowUp':
+        newHeading = this._prevHeading();
+        break;
+
+      case 'ArrowRight':
+      case 'ArrowDown':
+        newHeading = this._nextHeading();
+        break;
+
+      case 'Home':
+        newHeading = this._firstHeading();
+        break;
+
+      case 'End':
+        newHeading = this._lastHeading();
+        break;
+
+      case 'Enter':
+      case ' ':
+      case 'Spacebar':
+        this._expand(this._getClosestHeading(document.activeElement));
+        break;
+
+      // Any other key press is ignored and passed back to the browser.
+      default:
+        return;
+    }
+
+    event.preventDefault();
+    selectedEl.setAttribute('tabindex', '-1');
+    if (newHeading) {
+      newHeading.setAttribute('tabindex', '0');
+      newHeading.focus();
+    }
+  }
+
+  _expand (heading : Element) {
+
+    if (heading === undefined) {
+      return;
+    }
+
+    const content = heading.nextElementSibling;
+    
+    // heading elment should always have nextElementSibling (checked on _childrenChange)
+    // but in case it is null, return.
+    if (content === null) {
+      return;
+    }
+
+    // toggle expanded and aria-expanded attoributes
+    if (content.hasAttribute('expanded')) {
+      content.removeAttribute('expanded')
+      content.setAttribute('aria-expanded', 'false');
+    } else {
+      content.setAttribute('expanded', '') 
+      content.setAttribute('aria-expanded', 'true');  
+    }
+  }
+  
+  // children of multi-panel should always in even nuber (heading/content pair)
+  // if children are odd numbers, add a div at the end to prevent potential error.
+  _childrenChange () {
+    if (this.children.length % 2 !== 0) {
+      console.error('detected odd number of elements inside multi-panel, please make sure you have heading/content pair')
+    }
+
+    const children : Element[] = Array.from(this.children);
+    let tabindexAlreadySet : boolean = false;
+    let randomId : string;
+
+    children.forEach((child, i) => {
```

Comment:
I think this would be more readable as:

```js
for (let i = 0; i < children.length; i += 2) {
  const heading = children[i];
  const content = children[i+1];
  const randomId = Math.random().toString(36).substr(2, 9);

  // …
}
```

As you don't need to do the odd/even check, and `randomId` gets better scoping.

Reply:
handled this in suggested way.